### PR TITLE
Publish docker images to ghcr and make use of them

### PR DESCRIPTION
**Pull request by kasbah**
Making this pull against master, following on from #47

### CODE REVIEW

**Comment by kasbah** [comment]
Deployed this to https://staging.kitspace.org (using `docker-compose -f docker-compose.deploy.yml up`). Frontend image needs some work it seems, page is completely blank (not sure why). Let's see if your new image will fix it when it builds successfully.

File: frontend/Dockerfile.prod [hunk 1]
```diff
@@ -7,10 +7,10 @@ WORKDIR /app
 
 COPY package.json yarn.lock ./
 
-RUN yarn install
+RUN yarn --prune-lockfile
 
 COPY . .
 
-RUN yarn build
+RUN yarn build-fallback
```
Comment (kasbah) [inline frontend/Dockerfile.prod:14]: ???

**Comment by AbdulrhmnGhanem** [comment]
Exporting as a static fallback makes some pages don't work as expected for now.

**Comment by kasbah** [comment]
Re-deployed to https://staging.kitspace.org. Hmmmm

image

**Comment by AbdulrhmnGhanem** [comment]
This page was working fine locally! Is there anything in the log output?

**Comment by kasbah** [comment]
Nothing that I noticed. Try and deploy it yourself to abdo-dev.kitspace.org (I added processor.abdo-dev.kitspace.org to our DNS so that should work too).

**Comment by AbdulrhmnGhanem** [comment]
> Nothing that I noticed. Try and deploy it yourself to abdo-dev.kitspace.org (I added processor.abdo-dev.kitspace.org to our DNS so that should work too).

Same behavior. 

It doesn't print any logs locally also.

**Comment by AbdulrhmnGhanem** [comment]
Running the image by itself makes fixes the `not found` problems though some pages still don't work. Running by docker-compose breaks everything. I ran out of ideas, will leave it for a while and look at something else.

**Comment by AbdulrhmnGhanem** [comment]
Fixed it and re-deployed :rocket:

File: docker-compose.deploy.yml [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+    image: 'ghcr.io/kitspace/frontend-v2:latest'
+    ports:
+      - '1234:3000'
+    environment:
+      - KITSPACE_GITEA_URL=${KITSPACE_SCHEME}://gitea.${KITSPACE_DOMAIN}${KITSPACE_EXTERNAL_PORT}
+      - KITSPACE_PROCESSOR_URL=${KITSPACE_SCHEME}://processor.${KITSPACE_DOMAIN}${KITSPACE_EXTERNAL_PORT}
+    command: /bin/sh --verbose -c "yarn build && yarn start"
+    volumes:
+      - next:/app/.next/
+      - public:/app/public/
```
Comment (kasbah) [inline docker-compose.deploy.yml:13]: I don't think anything is changing in /public, shouldn't need a volume.

Comment (kasbah) [reply docker-compose.deploy.yml:13]: Nvm, I see it's a shared volume with Nginx now.

**Comment by AbdulrhmnGhanem** [comment]
Now, Gitea only gets built if there's a change in it that saves about 10 minutes, and most tests pass but merging master into this branch broke some stuff.
image

**Comment by AbdulrhmnGhanem** [comment]
I can squeeze another 3 minutes from the CI by hashing processor and only build it if the directory content changed. Should I do that or just leave it for now?

**Comment by kasbah** [comment]
Let's not over optimize. Leave it for now.


On January 30, 2021 at 3:59 AM, Ghanem (notifications@github.com) wrote:
I can squeeze another 3 minutes from the CI by hashing processor and only build it if the directory content changed. Should I do that or just leave it for now?

**Comment by AbdulrhmnGhanem** [comment]
@kasbah, I think this ready for review!

**Pull request merged**